### DISCUSSION TITLE
Format long specific non-location format in 2.0

### DISCUSSION THREAD

**Discussion by Nerixyz** [unicode-org/icu4x#5885] [Duration Formatter]
Hi, I was trying the new 2.0.0-beta to format some time zones. I want to format the time like this: `12:08:19 Mitteleuropäische Normalzeit`. I'm probably missing something, but it seems like that's not possible, neither with the `Combo<T, Zs>` (note that `Combo<T, Z>` can't be constructed) nor with the `CompositeFieldSet::TimeZone`:

```rust
use icu_datetime::{
    fieldsets::{
        enums::{CompositeFieldSet, TimeFieldSet, ZoneStyle},
        T, Z,
    },
    DateTimeFormatter,
};
use icu_locale_core::locale;
use icu_timezone::IxdtfParser;

fn main() {
    let zdt = IxdtfParser::new()
        .try_from_str("2024-08-08T12:08:19+01:00[Europe/Berlin]")
        .unwrap();

    let formatter = DateTimeFormatter::try_new(
        locale!("de").into(),
        CompositeFieldSet::TimeZone(TimeFieldSet::T(T::long()), ZoneStyle::Z),
    )
    .unwrap();
    dbg!(formatter.format_any_calendar(&zdt).to_string()); // 12:08:19 MEZ

    let formatter =
        DateTimeFormatter::try_new(locale!("de").into(), T::long().zone_z()).unwrap();
    dbg!(formatter.format_any_calendar(&zdt).to_string()); // 12:08:19 MEZ

    let formatter = DateTimeFormatter::try_new(locale!("de").into(), Z::long()).unwrap();
    dbg!(formatter.format_any_calendar(&zdt).to_string()); // Mitteleuropäische Normalzeit
}
```

As you can see, it's only possible to format the short specific non-location style, not the long one, when combined with a time.

Aside: the `CompositeFieldSetSerde` (experimental) specifies `FieldSetField::Zone{Generic,Specific}{Short,Long}` but they aren't used.
Aside2: the [tests for the lengths](https://github.com/unicode-org/icu4x/blob/8717dfc98191fae0f228f63d9d5c337d723ffcad/components/datetime/tests/datetime.rs#L512-L523) are all skipped because [none of them specify a semantic skeleton](https://github.com/unicode-org/icu4x/blob/8717dfc98191fae0f228f63d9d5c337d723ffcad/components/datetime/tests/datetime.rs#L340).

**Comment by sffc** [comment]
Good question!

There's code that always sets the time zone length to Short when it is combined with other fields:

https://github.com/unicode-org/icu4x/blob/8717dfc98191fae0f228f63d9d5c337d723ffcad/components/datetime/src/raw/neo.rs#L592

This was done to reduce data size; however, I don't see it mentioned in UTS 35. And, even if we do this by default, there should at least be a way to get the other behavior (long time zone name).

The fix would involve adding support for `Combo<T, Z>`.
